File: server/api/routes/sitemap_routes.py
from fastapi import APIRouter
from controllers.sitemap_controller import (
    generate_sitemap
)

router = APIRouter()

@router.get("/sitemap")
async def fetch_sitemap():
    return await generate_sitemap()

@router.get("/sitemap.xml")
async def fetch_sitemap_xml():
    return await generate_sitemap()

File: server/api/routes/sitemapRoutes.js
const express = require('express');
const { generateSitemap, generateProductSitemap } = require('../controllers/sitemapController.js');

const router = express.Router();

// GET /api/sitemap.xml - Generate main sitemap
router.get('/sitemap.xml', generateSitemap);

// GET /api/sitemap-products.xml - Generate products-specific sitemap
router.get('/sitemap-products.xml', generateProductSitemap);

module.exports = router;
